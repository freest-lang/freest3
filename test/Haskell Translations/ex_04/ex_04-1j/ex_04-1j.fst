-- IV exercise 1j

data IntList = End | List Int IntList

posicoes : IntList -> Int -> IntList
posicoes list m = posicoes' list m 0

posicoes' : IntList -> Int -> Int -> IntList
posicoes' list m i =
    case list of {
        End -> End,
        List x rest -> if mod x m == 0 
                            then List i (posicoes' rest m (i+1))
                            else posicoes' rest m (i+1)
    }

main : IntList
main = posicoes (List 1 (List 3 (List 6 (List 2 (List 5 (List 15 (List 3 (List 5 (List 7 (List 18 End)))))))))) 3
-- result = List 1 (List 2 (List 5 (List 6 (List 9 End))))