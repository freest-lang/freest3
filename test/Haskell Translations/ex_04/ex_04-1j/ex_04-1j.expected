List 1 (List 2 (List 5 (List 6 (List 9 End))))